### ISSUE TITLE
Sardine.put on a locked resource

### ISSUE THREAD

**Issue by lasselindqvist** [lookfirst/sardine#368]
It would be useful if Sardine could do a put call on a locked resource.
Now there is no way to pass the lockToken to the put call and this the user flow of
1. sardine.lock
2. sardine.put // perhaps multiple calls
3. sardine.unlock

is not possible.

Sardine.put has many overloads and many methods that take Strings as input.
Might be worth considering turning the Sardine.lock method into a method that returns a Lock-object. (new interface). It could have a now just a one method, getToken(), but could also support getExpiration etc. methods in future. (There is of course the XML class Lockinfo which we might not want to directly expose for the client)

Then we could add Sardine.put(String url, Lock lock) method for calling put with a lock.

**Comment by dkocher** [comment]
This is possible as of 73ccb7ec which allows to pass any additional headers.

**Comment by lasselindqvist** [comment]
Yes but it is not very easy for a user to use. They would need to manually add the lock metadata there.

**Comment by lookfirst** [comment]
Hi @lasselindqvist, as David says, Sardine allows this pretty easily by allowing you to pass in whatever headers you need. If you'd like an easier interface, you're welcome to wrap yours around the core library. Adding more methods and overloads for this one feature doesn't seem applicable, but I'm open to further discussion on reasoning.
